src/format/ProtonPassReader.cpp

```diff
@@ -0,0 +1,221 @@
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ProtonPassReader.h"
+
+#include "core/Database.h"
+#include "core/Entry.h"
+#include "core/Group.h"
+#include "core/Metadata.h"
+#include "core/Tools.h"
+#include "core/Totp.h"
+#include "crypto/CryptoHash.h"
+
+#include <QFileInfo>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonParseError>
+#include <QMap>
+#include <QScopedPointer>
+#include <QUrl>
+
+namespace
+{
+    Entry* readItem(const QJsonObject& item)
+    {
+        const auto itemMap = item.toVariantMap();
+        const auto dataMap = itemMap.value("data").toMap();
+        const auto metadataMap = dataMap.value("metadata").toMap();
+
+        // Create entry and assign basic values
+        QScopedPointer<Entry> entry(new Entry());
+        entry->setUuid(QUuid::createUuid());
+        entry->setTitle(metadataMap.value("name").toString());
+        entry->setNotes(metadataMap.value("note").toString());
+
+        if (itemMap.value("pinned").toBool()) {
+            entry->addTag(QObject::tr("Favorite", "Tag for favorite entries"));
+        }
+
+        // Handle specific item types
+        auto type = dataMap.value("type").toString();
+
+        // Login
+        if (type.compare("login", Qt::CaseInsensitive) == 0) {
+            const auto loginMap = dataMap.value("content").toMap();
+            entry->setUsername(loginMap.value("itemUsername").toString());
+            entry->setPassword(loginMap.value("password").toString());
+            if (loginMap.contains("totpUri")) {
+                auto totp = loginMap.value("totpUri").toString();
+                if (!totp.startsWith("otpauth://")) {
+                    QUrl url(QString("otpauth://totp/%1:%2?secret=%3")
+                                 .arg(QString(QUrl::toPercentEncoding(entry->title())),
+                                      QString(QUrl::toPercentEncoding(entry->username())),
+                                      QString(QUrl::toPercentEncoding(totp))));
+                    totp = url.toString(QUrl::FullyEncoded);
+                }
+                entry->setTotp(Totp::parseSettings(totp));
+            }
+
+            if (loginMap.contains("itemEmail")) {
+                entry->attributes()->set("login_email", loginMap.value("itemEmail").toString());
+            }
+
```

Comment:
Since you can also import into an existing database, I'd prefer if all imported items had a collision check, not just the additional fields. That actually applies to the other importers as well.

Reply:
Not sure i understand? All imported items are new entries.

Reply:
Yah, never mind, temporary brain fart. I got somehow hung up on the entry name, but adding a new entry with unique UUID should not cause any issues.